src/test.js

```diff
@@ -1,5 +1,5 @@
 /* eslint-disable no-undef */
-const StatusScreen = require("./status/index").Screen;
+const HomeScreen = require("./screens/homeScreen/HomeScreen").Screen;
```

Comment:
Just trying to think of a way to improve the current name of `test.js`, since that is pretty general and not all the tests live here.

I see that this prepares the test runner and I see that navigation is being handled. Should (can) those 2 things be separated and file names clarify that distinction?

Reply:
Ohh i'm glad you mentioned this. I sat staring at this and utilities.js for a while as well as properties.js. thinking there needs to be a way to better organize this. I completely agree with you. Right now on my local machine no tests live there anymore. So what is this? and how is it different from utilities.js and properties.js?

As far as my brain can tell
-test. js: 
exports class Test -> includes setup of loop (i.e. closed loop/open loop, language, data, ect.
This is used before every single test to prepare the loop environment for testing.
Then, as you said before, there's a ton of functions in there for navigation, which the setup uses, but are NOT exclusive to the setup.
- I think they should be pulled apart. that's my first instinct. Perhaps something like LoopSetup and utilities for the functions,...but wait we already have a utilities! so what's that:

utilities.js:
- another class being exported called Utilities(this is why its helpful to do class file naming in pascal, otherwise you don't know what's a class and what's not at a glance)
- this is a bunch of helper functions for common tasks. like addConfiguredPump(), or addCarbohydrates()
- this makes sense to me for the most part. I'm not entirely sure why it's a class and not just a bunch of exported functions, but maybe it was to make the importing/exporting easier? not sure...

So, what does this mean so far? that means if we keep utilities.js as is, and we rename test.js to something like LoopSetup.js and keep just the test preparation in there, we still have all this navigation stuff with no real home...However, a thought occurs to me, why is the specific screen navigation living outside of their screens? why doesn't ```
async OpenHomeScreen() {
    return this.homeScreen;
  }
```
live inside the HomeScreen? My thought is that is makes sense there. Then if you wanted to use that screen in your test...you just import the HomeScreen...or the PredictedGlucoseScreen..or whatever screen you're jimmying around with at the time.

SO that sounds like a plan, but wait we forgot some other friends that live in this directory...

properties.js - mapping screens to their settings. I'm not entirely sure of the full scope of what's going on here, honestly...i think i'll figure it out as I break things.

index.js - exports two classes from their files. not necessary and will be removed
action.js - specific detox actions, condensed and made easy to use in tests. i'm a fan.
match - detox matchers. it's an attempt to condense the detox element matching code. I see the use for this, so it's fine, we just have to make sure that whenever you want to match an element you have to use something out of this file and not straight from the detox docs. otherwise there would be a lot of code duplication. This can be handled in an updated ReadMe.

So, yep. This was a huge brain dump. let me know what you think

Reply:
wow. i also realized that in the base folder there is screen.js ...which seems to be more helper functions and for example it has   

    async SwipeDownUntilVisible(labelToSee) {
        await action.SwipeDownUntilVisible(labelToSee);
      }

but actions.js has the same thing

    async SwipeDownUntilVisible(desiredLabel) {
            await _swipeUntilVisible(desiredLabel, 'down');
        },
        
   Need to see why that duplication is there. if it's not necessary then all duplicated actions should be removed and should live inside the actions.js